Resources/Catalog/FCConditionParser.py

```diff
@@ -0,0 +1,312 @@
+from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
+
+
+
+class FCConditionParser(object):
+  """
+    This objects allows to evaluate conditions on whether or not
+    a given operation should be evaluated on a given catalog
+    for a given lfn (be glad so many things are given to you !).
+
+    The conditions are expressed as boolean logic, where the basic bloc has
+    the form "pluginName=whateverThatWillBePassedToThePlugin".
+    The basic blocs will be evaluated by the respective plugins, and the result can
+    be combined using the standard boolean operators:
+      * ! for not
+      * & for and
+      * | for or
+      * [ ] for prioritizing the operations
+
+    All these characters, as well as the '=' symbol cannot be used in any expression to be
+    evaluated by a plugin.
+
+    The rule to evaluate can either be given at calling time, or can be retrieved
+    from the CS depending on the context (see doc of __call__ and __getConditionFromCS)
+
+
+    Example of rules are:
+
+      Filename=startswith('/lhcb') & Proxy=voms.has(/lhcb/Role->production)
+      [Filename=startswith('/lhcb') & !Filename=find('/user/')] | Proxy=group.in(lhcb_mc, lhcb_data)
+
+  """
+  
+  
+  # Some characters are reserved for the grammar
+  __forbidenChars = ( '[', ']', '!', '&', '|', '=' )
+  __allowedChars = ''.join( set( printables ) - set( __forbidenChars ) ) + ' '
+
+
+  # Defines the basic shape of a rule : pluginName=whateverThatWillBePassedToThePlugin
+  __pluginOperand = Word( __allowedChars ) + Literal( '=' ) + Word( __allowedChars )
+
+
+  # define classes to be built at parse time, as each matching
+  # expression type is parsed
+
+  # Binary operator base class
+  class _BoolBinOp( object ):
+    """ Abstract object to represent a binary operator
+
+    """
+
+    reprsymbol = None  # Sign to represent the boolean operation
+
+    # This is the boolean operation to apply
+    # Could be None, but it should be callable
+    evalop = lambda _x : None
+
+    def __init__( self, token ):
+      """
+           :param token : the token matching a binary operator
+                         it is a list with only one element which itself is a list
+                         [ [ Arg1, Operator, Arg2] ]
+                         The arguments themselves can be of any type, but they need to
+                         provide an "eval" method that takes **kwargs as input,
+                         and return a boolean
+      """
+
+      # Keep the two arguments
+      self.args = token[0][0::2]
+
+    def __str__( self ):
+      """ String representation """
+
+      sep = " %s " % self.reprsymbol
+      return "(" + sep.join( map( str, self.args ) ) + ")"
+
+
+
+    def eval( self, **kwargs ):
+      """ Perform the evaluation of the boolean logic
+          by applying the operator between the two arguments
+
+          :param **kwargs whatever information is given to plugin (typically lfn)
```

Comment:
This :param **kwargs needs to end with a colon.
I.e.:
```
:param **kwargs:
```
There are more cases of missing colons for :param and :return(s).
Otherwise things are not rendered correctly (e.g., http://diracgrid.org/files/docs/CodeDocumentation/Core/Utilities/ReturnValues.html)